pkg/bucketeer/cache/processor/segment_user_processor.go

```diff
@@ -0,0 +1,197 @@
+	Run()
+
+	// Close tears down all Processor activities, after ensuring that all events have been delivered.
+	Close()
+}
+
+type segmentUserProcessor struct {
+	cache             cache.Cache
+	segmentUsersCache cache.SegmentUsersCache
+	pollingInterval   time.Duration
+	apiClient         api.Client
+	closeCh           chan struct{}
+	loggers           *log.Loggers
+}
+
+// ProcessorConfig is the config for Processor.
+type SegmentUserProcessorConfig struct {
+	// Cache
+	Cache cache.Cache
+
+	// PollingInterval is a interval of polling segment users from the server
+	PollingInterval time.Duration
+
+	// APIClient is the client for Bucketeer service.
+	APIClient api.Client
+
+	// Loggers is the Bucketeer SDK Loggers.
+	Loggers *log.Loggers
+}
+
+const (
+	segmentUserCacheTTL        = time.Duration(0)
+	segmentUsersRequestedAtKey = "bucketeer_segment_users_requested_at"
+)
+
+func NewSegmentUserProcessor(conf *SegmentUserProcessorConfig) SegmentUserProcessor {
+	p := &segmentUserProcessor{
+		cache:             conf.Cache,
+		segmentUsersCache: cache.NewSegmentUsersCache(conf.Cache),
+		pollingInterval:   conf.PollingInterval,
+		apiClient:         conf.APIClient,
+		closeCh:           make(chan struct{}),
+		loggers:           conf.Loggers,
+	}
+	return p
+}
+
+func (p *segmentUserProcessor) Run() {
+	go p.runProcessLoop()
+}
+
+func (p *segmentUserProcessor) Close() {
+	p.closeCh <- struct{}{}
+}
+
+func (p *segmentUserProcessor) runProcessLoop() {
+	defer func() {
+		p.loggers.Debug("bucketeer/cache: segmentUsers runProcessLoop done")
+	}()
+	ticker := time.NewTicker(p.pollingInterval)
+	defer ticker.Stop()
+	// Update the cache once when starts polling
+	if err := p.updateCache(); err != nil {
+		p.loggers.Errorf("bucketeer/cache: segmentUsers failed to update segment users cache. Error: %v", err)
+	}
+	for {
+		select {
+		case <-ticker.C:
+			if err := p.updateCache(); err != nil {
+				p.loggers.Errorf("bucketeer/cache: segmentUsers failed to update segment users cache. Error: %v", err)
+				continue
+			}
+		case <-p.closeCh:
+			return
+		}
+	}
+}
+
+func (p *segmentUserProcessor) updateCache() error {
+	ftsID, err := p.segmentUsersCache.GetSegmentIDs()
+	if err != nil {
+		if !errors.Is(err, cache.ErrNotFound) {
+			return err
+		}
+		p.loggers.Debug("bucketeer/cache: segmentUsers updateCache segment IDs not found")
+	}
+	requestedAt, err := p.getSegmentUsersRequestedAt()
+	if err != nil {
+		if !errors.Is(err, cache.ErrNotFound) {
+			return err
+		}
+		p.loggers.Debug("bucketeer/cache: segmentUsers updateCache requestedAt not found")
+	}
+	req := model.NewGetSegmentUsersRequest(
+		ftsID,
+		requestedAt,
+	)
+	// Get the latest cache from the server
+	resp, size, err := p.apiClient.GetSegmentUsers(req)
+	if err != nil {
+		return err
+	}
+	p.loggers.Debugf("bucketeer/cache: GetSegmentUsers response: %v, size: %d", resp, size)
+	// Delete all the local cache and save the new one
+	if resp.ForceUpdate {
```

Comment:
If the `ForceUpdate` is true, it must delete all the local cache and save the new one from the response.